modules/pubsub/src/main/java/com/spotify/ffwd/pubsub/PubsubPluginSink.java

```diff
@@ -116,7 +121,18 @@ public void onSuccess(String messageId) {
 
     try {
       final ByteString m = ByteString.copyFrom(serializer.serialize(metrics, writeCache));
-      publishPubSub(m);
+
+      if (m.size() > MAX_BATCH_SIZE_BYTES) {
+        logger.debug("Above byte limit, resizing batch", metrics.size());
+        int times = (int)Math.ceil(m.size()/MAX_BATCH_SIZE_BYTES);
```

Comment:
Would `(m.size() / MAX_BATCH_SIZE_BYTES) + (m.size() % MAX_BATCH_SIZE_BYTES)` work/potentially be bit cleaner?

Reply:
And then could make the constant an int or long